### PR TITLE
Update low level errors nodes edgesets

### LINKED ISSUE #146: Update low-level error checking and messages
Error messages in the low-level C code still refer to coalescence records. Update these symbolic constants with better names and update the error messages.

### PR DESCRIPTION
**Pull request by ashander**
Updates error codes to refer to nodes/edgesets versus coalescence records

closes #146

### CODE REVIEW

File: lib/msprime.c [hunk 1]
```diff
@@ -133,17 +133,16 @@ msp_strerror(int err)
         case MSP_ERR_BAD_RECOMBINATION_MAP:
             ret = "Bad recombination map provided.";
             break;
-        case MSP_ERR_BAD_COALESCENCE_RECORDS_SAMPLE_SIZE:
-            ret = "Bad coalescence records in file: Samples must be labeled 0...n-1 and not be internal nodes.";
+        case MSP_ERR_BAD_NODES_SAMPLE_SIZE:
+            ret = "Bad nodes in file: No sampled nodes.";
```
Comment (jeromekelleher) [inline lib/msprime.c]: Technically we need at least 2 samples I think, so maybe we should change this to ``MSP_ERR_INSUFFICIENT_SAMPLES`` and change the error message? Probably the "Bad nodes in file" bit doesn't add much and can be removed.

Comment (ashander) [inline lib/msprime.c]: :+1:

**Comment by ashander** [comment]
Thanks. I did that fixed and added a test for one of these. I also tried to write a test for samples < 2 but it didn't work as expected (so commented out)

**Comment by jeromekelleher** [comment]
Your test case was perfect @ashander, there was a bug in the code. I've fixed this up and added another check (makes sure that samples are 0...n - 1, which we're still depending on in a few places).  

Thanks again!

File: lib/tree_sequence.c [hunk 2]
```diff
@@ -719,10 +719,18 @@ tree_sequence_init_nodes(tree_sequence_t *self)
             self->sample_size++;
         }
     }
-    if (self->nodes.num_records > 0 && self->sample_size == 0) {
+    if (self->nodes.num_records > 0 && self->sample_size < 2) {
```
Comment (ashander) [inline lib/tree_sequence.c]: Ah! Oops no wonder that test didn't work. Thanks

**Pull request merged**